src/compiler/checker.ts

```diff
@@ -794,12 +794,17 @@ namespace ts {
             // 2. inside a function
             // 3. inside an instance property initializer, a reference to a non-instance property
             // 4. inside a static property initializer, a reference to a static method in the same class
+            // 5. inside a TS export= declaration (since we will move the export statement during emit to avoid TDZ)
             // or if usage is in a type context:
             // 1. inside a type query (typeof in type position)
-            if (usage.parent.kind === SyntaxKind.ExportSpecifier) {
+            if (usage.parent.kind === SyntaxKind.ExportSpecifier || (usage.parent.kind === SyntaxKind.ExportAssignment && (usage.parent as ExportAssignment).isExportEquals)) {
                 // export specifiers do not use the variable, they only make it available for use
                 return true;
             }
+            // When resolving symbols for exports, the `usage` location passed in can be the export site directly

```

Comment:
Ehh, I don't really like the idea of special casing it in `checkResolvedBlockScopedVariable`, since it's only looking up the declaration associated with a symbol and handing it off; it'd have to have the same special-casey logic as here (well, it'd have to dive down rather than walk up). Except we already to special-case things here, and not in `checkResolvedBlockScopedVariable`.